software/gatk4/fastqtosam/main.nf

```diff
@@ -0,0 +1,53 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process GATK4_FASTQTOSAM {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::gatk4=4.2.0.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/gatk4:4.2.0.0--0"
+    } else {
+        container "quay.io/biocontainers/gatk4:4.2.0.0--0"
+    }
+
+    input:
+    tuple val(meta), path(reads)
+
+    output:
+    tuple val(meta), path("*.bam"), emit: bam
+    path "*.version.txt"          , emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    if (meta.single_end) {
+        """
+        gatk FastqToSam \\
+            -F1 $reads \\
+            -O ${prefix}.bam \\
+            -SM $prefix \\
+            $options.args
+
+        gatk --version | grep Picard | sed "s/Picard Version: //g" > ${software}.version.txt
+        """
+    } else {
+        """
+        gatk FastqToSam \\
+            -F1 ${reads[0]} \\
+            -F2 ${reads[1]} \\
+            -O ${prefix}.bam \\
+            -SM $prefix \\
+            $options.args
+
+        gatk --version | grep Picard | sed "s/Picard Version: //g" > ${software}.version.txt
+        """
+    }
```

Comment:
```suggestion
    def software   = getSoftwareName(task.process)
    def prefix     = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
    def read_files = meta.single_end ? "-F1 $reads" : "-F1 ${reads[0]} -F2 ${reads[1]}"
    """
    gatk FastqToSam \\
        $read_files \\
        -O ${prefix}.bam \\
        -SM $prefix \\
        $options.args
    gatk --version | grep Picard | sed "s/Picard Version: //g" > ${software}.version.txt
    """
```

Reply:
How about this instead @ntoda03 ?

Reply:
@drpatelh Ah yes that's a smart way to do it